src/server/handlers/store.js

```diff
@@ -101,3 +101,90 @@ export const signAgreement = async (req, res) => {
   const json = await response.json();
   return res.status(response.status).send(json);
 };
+
+export const nameOwnership = async (req, res) => {
+  const snapName = req.query.snap_name;
+  const root = req.query.root;
+  const discharge = req.query.discharge;
```

Comment:
I know that various things in this file are already receiving root and discharge macaroons, but they're all thin shims over store APIs rather than containing much logic of their own, and they can be replaced with direct calls to the store from the client side once the store sets the appropriate CORS headers.  This method is different since it contains non-trivial logic of its own.

So, could you move all this to the client side to avoid building up more technical debt that we'll need to rearrange once #272 is done?  All the server-side facilities are already in place: `/dev/api/acl/` can be called directly using CORS, and `/dev/api/register-name/` already has a shim in this file.